cdc/log.cc

```diff
@@ -1463,22 +1463,25 @@ class transformer final : public change_processor {
 
         assert(_builder);
 
-        auto image_ck = _builder->allocate_new_log_row(op);
-        if (ck) {
-            _builder->set_clustering_columns(image_ck, *ck);
-        }
-
         const auto kind = ck ? column_kind::regular_column : column_kind::static_column;
 
         cell_map* row_state;
         if (ck) {
             row_state = get_row_state(_clustering_row_states, *ck);
-            if (!row_state) {
+            if (!row_state && op == cdc::operation::pre_image) {
                 // We have no data for this row, we can stop here
                 return;
             }
         } else {
             row_state = &_static_row_state;
+            if (row_state->empty()) {
+                return;
+            }
+        }
+
+        auto image_ck = _builder->allocate_new_log_row(op);
```

Comment:
Why is this line not inside the if()? It's not used outside the if, is it?
Or maybe you did this way to ensure it lives a long enough time? But is is a long enough time? I'm afraid I am not familiar with this code enough to understand.

Reply:
? The code path is used from two true-branches (i.e. both cases where we _don't shortcircuit image generation).

Reply:
I meant, as far as I can tell (maybe I don't see the entire code so I'm missing something), this "image_ck" variable is only used inside the if() below, so why isn't it defined inside the if?

Reply:
The var is used in lambda capture three lines below. It is at the right scope.